autoload/go/highlight_test.vim

```diff
@@ -104,21 +104,45 @@ function! Test_numeric_literal_highlight() abort
   let tests = {
         \ 'lone zero': {'group': 'goDecimalInt', 'value': '0'},
         \ 'integer': {'group': 'goDecimalInt', 'value': '1234567890'},
+        \ 'integerGrouped': {'group': 'goDecimalInt', 'value': '1_234_567_890'},
+        \ 'integerErrorLeadingUnderscore': {'group': 'goDecimalErrorInt', 'value': '_1234_567_890'},
+        \ 'integerErrorTrailingUnderscore': {'group': 'goDecimalErrorInt', 'value': '1_234_567890_'},
+        \ 'integerErrorDoubleUnderscore': {'group': 'goDecimalErrorInt', 'value': '1_234__567_890'},
+        \ 'integerErrorDoubleTrailingUnderscore': {'group': 'goDecimalErrorInt', 'value': '1_234__567_890__'},
```

Comment:
this test doesn't add value over the `integerErrorDoubleUnderscore`, because this test also has the double underscores in the middle of the number.

Reply:
Yeah thanks, nice catch! Fixed. Technically it works as expected because we have a separate group for matching trailing underscores. But I completely agree with you, to won't get in false-positives.